Packs/AtlassianConfluenceCloud/Integrations/AtlassianConfluenceCloud/AtlassianConfluenceCloud.py

```diff
@@ -1333,14 +1379,101 @@ def confluence_cloud_group_list_command(client: Client, args: Dict[str, str]) ->
         raw_response=response_json)
 
 
+def fetch_events(client: Client, last_run: dict[str, Any], fetch_limit: int):
+    demisto.debug(f'Starting fetch_events with last_run: {last_run} and fetch_limit: {fetch_limit}')
+    end_date = int((time.time() - 5) * 1000)
+    last_end_date = last_run.get('end_date', 0)
+    start_date = last_end_date + 1 if last_end_date else end_date - 60000
+    next_link = last_run.get('next_link', '')
+    is_cleanup = bool(next_link)
+    total_length = 0
+
+    demisto.debug(f'Initial start_date: {start_date}, end_date: {end_date}, next_link: {next_link}, is_cleanup: {is_cleanup}')
+    while True:
+        response = run_fetch_mechanism(client, next_link, start_date, end_date)
+        events = response['results']
+        next_link = response['_links'].get('next', None)
+        total_length += len(events)
+
+        demisto.debug(f'Fetched {len(events)} events, total_length: {total_length}, next_link: {next_link}')
+        if total_length >= fetch_limit:
```

Comment:
cant we pass this limit into the fetch? why truncate here in the code?

Reply:
limit passed to API is the limit on page size, not the overall limit. They do not provide that parameter